### PR TITLE
test: better `talosctl ls` tests

### PR DESCRIPTION
**Pull request by AlekSi**
Refs #3018.

Actual fixes will follow.

- [x] you linked an issue (if applicable)
- [x] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

### CODE REVIEW

File: internal/integration/base/run.go [hunk 1]
```diff
@@ -44,13 +44,6 @@ func ShouldFail() RunOption {
 	}
 }
 
-// ShouldSucceed tells Run command should succeed (that is default).
-func ShouldSucceed() RunOption {
-	return func(opts *runOptions) {
-		opts.shouldFail = true
```
Comment (AlekSi) [inline internal/integration/base/run.go:50]: Not used, and there is a typo (it should be `false`)

File: hack/test/e2e-docker.sh [hunk 2]
```diff
@@ -29,11 +29,12 @@ function create_cluster {
...
-  "${TALOSCTL}" cluster destroy --name "${CLUSTER_NAME}"
+  "${TALOSCTL}" cluster destroy --name "${CLUSTER_NAME}" --provisioner "${PROVISIONER}"
 }
 
 create_cluster
 get_kubeconfig
 ${KUBECTL} config set-cluster e2e-docker --server https://10.5.0.2:6443
 run_talos_integration_test_docker
 run_kubernetes_integration_test
+destroy_cluster
```
Comment (smira) [inline hack/test/e2e-docker.sh:40]: I think this is wrong, as I believe we rely on docker cluster to be up in CAPI tests - e2e-docker creates a cluster, e2e-capi installs CAPI, e2e-aws/gcp use that cluster to provision talos in the cloud

**Review by Unix4ever — DISMISSED** [review]

File: internal/integration/cli/list.go [hunk 3]
```diff
@@ -31,6 +33,79 @@ func (suite *ListSuite) TestSuccess() {
...
 }
 
+// TestDepth tests various combinations of --recurse and --depth flags.
+func (suite *ListSuite) TestDepth() {
+	suite.T().Parallel()
+
+	node := suite.RandomDiscoveredNode(machine.TypeJoin)
+
+	runAndParse := func(flags ...string) []string {
+		args := append([]string{"list", "--nodes", node, "/etc/kubernetes"}, flags...)
```
Comment (smira) [inline internal/integration/cli/list.go:43]: my concern with `/etc/kubernetes` is that it's managed by the kubelet, and its contents might change with `kubelet` version bumps or might change after some interaction with the kubelet

not sure which other directory is better, but my guess `/system/secrets` might be better, as at least it's managed by Talos and its contents are fixed (for control plane nodes only)

**Review by smira — APPROVED** [review]
Nice!

**Pull request merged**